examples/systemcalls/systemcalls.c

```diff
@@ -92,7 +118,31 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
-
+    int fd = open(outputfile, O_WRONLY | O_CREAT, S_IRWXG | S_IRWXO | S_IRWXU);
+    
+    pid_t child_process_id;
+    int wait_status;
+    
+    child_process_id = fork();
+    if (child_process_id == -1){
+        close(fd);
+        return false;
+    }
+    if (child_process_id == 0){                                 // Execute child process //
+        dup2(fd, 1);
+        close(fd);
+        if (execv(command[0], command) == -1){
+            exit(EXIT_FAILURE);
+        }
+    }
+    else{                                                       // Execute parent process //
+        do {
+            if (waitpid(child_process_id, &wait_status, 0) ==-1)
+                return false;
```

Comment:
Missing cleanup in these return paths.
Also see discussion in class about avoiding duplication of logic on these two very similar functions.